include/aspect/newton.h

```diff
@@ -61,6 +61,16 @@ namespace aspect
    * A Class which can declare and parse parameters and creates
    * material model outputs for the Newton solver.
    */
+
+  enum class PreconditionerStabilization
+  {
+    SPD, PD, symmetric, none
```

Comment:
I think there is an easier way: declare it like this:
```
  none = 0,
  symmetric = 1;
  PD = 2;
  SPD = symmetric | PD;
```
You can then in the appropriate place ask whether a particular bit is set.

Same for the enum below.